### DISCUSSION TITLE
Enterprise features for free: yes or no

### DISCUSSION THREAD

**Discussion by GajulanIrcem** [dani-garcia/vaultwarden#4132] [Q&A]
Since Bitwarden is open source and it is possible to self-host "[the entire Bitwarden infrastructure stack](https://bitwarden.com/open-source/)", I I thought it was also possible to use Enterprise features for free (as long as the instance is self-hosted). Isn't that the case? Am I misunderstanding things and every user needs to have a license anyway?

**Comment by jjlin** [comment]
They don't explain it clearly everywhere, but Bitwarden is not fully open source. Most of their enterprise features are governed by a commercial "source-available" license; see https://github.com/bitwarden/server/blob/master/LICENSE_FAQ.md.

Self-hosting their stack is no different from using their cloud version as far as license requirements go, even if you aren't using any of their enterprise features. I've seen some projects out there that strip out the licensing checks (e.g., [BitBetter](https://github.com/jakeswenson/BitBetter) and [VaultLibre](https://github.com/Ayitaka/VaultLibre)), but I don't have personal experience with them, and while I'm not a lawyer, it seems almost certain that they're breaking the terms of Bitwarden's licenses.

**Reply by GajulanIrcem** [reply]
thank you for your reply.
So, does this mean that Enterprise features are available for free with vaultwarden?

**Reply by jjlin** [reply]
Some, but not all. See https://github.com/dani-garcia/vaultwarden/wiki for details.
